src/mock.ts

```diff
@@ -91,57 +101,68 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
   // and completes the customMock object with any fields
   // defined on genericMock
   // only merges objects or arrays. Scalars are returned as is
-  function mergeMocks(genericMockFunction, customMock) {
+  function mergeMocks(genericMockFunction: IMockFn, customMock: any): any {
     if (Array.isArray(customMock)) {
-      return customMock.map((el) => mergeMocks(genericMockFunction, el));
+      return customMock.map((el: any) => mergeMocks(genericMockFunction, el));
     }
     if (isObject(customMock)) {
       return mergeObjects(genericMockFunction(), customMock);
     }
     return customMock;
   }
 
-  function assignResolveType(type) {
+  function getResolveType(namedFieldType: GraphQLNamedType) {
+    if ( (namedFieldType instanceof GraphQLInterfaceType) ||
+         (namedFieldType instanceof GraphQLUnionType) ) {
+        return namedFieldType.resolveType;
+    } else {
+        return undefined;
+    }
+  }
+
+  function assignResolveType(type: GraphQLType) {
     const fieldType = getNullableType(type);
     const namedFieldType = getNamedType(fieldType);
 
-    const oldResolveType = namedFieldType.resolveType;
+    const oldResolveType = getResolveType(namedFieldType);
     if (preserveResolvers && oldResolveType && oldResolveType.length) {
       return;
     }
 
     if (namedFieldType instanceof GraphQLUnionType ||
-        namedFieldType instanceof GraphQLInterfaceType
-    ) {
+        namedFieldType instanceof GraphQLInterfaceType) {
       // the default `resolveType` always returns null. We add a fallback
       // resolution that works with how unions and interface are mocked
-      namedFieldType.resolveType = (data, context, info) =>
-        info.schema.getType(data.typename);
+      namedFieldType.resolveType = (data: any, context: any, info?: GraphQLResolveInfo) => {
+        if ( undefined === info ) {
+            throw new Error(`no info provided for resolveType`);
+        }
+        return info.schema.getType(data.typename) as GraphQLObjectType;
+      };
     }
   }
 
-  const mockType = function mockType(type, typeName, fieldName) {
+  const mockType = function mockType(type: GraphQLType, typeName?: string, fieldName?: string): IResolveFn {
     // order of precendence for mocking:
     // 1. if the object passed in already has fieldName, just use that
     // --> if it's a function, that becomes your resolver
     // --> if it's a value, the mock resolver will return that
     // 2. if the nullableType is a list, recurse
     // 2. if there's a mock defined for this typeName, that will be used
     // 3. if there's no mock defined, use the default mocks for this type
-    return (...args) => {
-      const [root, queryArgs, context, info] = args;
-
+    return (root: any, args: { [key: string]: any }, context: any, info: GraphQLResolveInfo): any => {
```

Comment:
This is the same signature as a GraphQL resolve function. Maybe we can import those typings from graphql and use them here?